ComplianceAsCode/content
# Add script to update generated SDS to SCAP 1.3

#### Description:

- Add script that updates generated DataStream to be compliant with SCAP 1.3.

#### How to test

- Use [scapval-1.3.2](https://csrc.nist.gov/Projects/Security-Content-Automation-Protocol/SCAP-Releases/SCAP-1-3)
  - Use `openjdk-1.8.0` on a RHEL environment. (for some reason it doesn't work on Fedora)
  - Increase allocated memory: `-Xmx1024m`
- Workaround bugs in scap-schematron-rules-1.3.2 (just to test that scapval passes):
  - `group_id` and `user_id` cannot be zero, tracked here https://github.com/OVAL-Community/OVAL/issues/23
    - workaround it : 
      - `sed -i -E "s/(group_id .*\">)/\1-1/" ./ssg-rhel7-ds.xml`
      - `sed -i -E "s/(user_id .*\">)/\1-1/" ./ssg-rhel7-ds.xml`
  - OCIL checks for `security_patches_up_to_date` is causing fail of SRC-377, when requirement is about OVAL checks. I have e-mailed NIST about this to try to sort out what is the issue.
    - workaround it:
      - `awk '/check-content-ref.*up_to_date_ocil/{for(x=NR-1;x<=NR+1;x++)d[x];}{a[NR]=$0}END{for(i=1;i<=NR;i++)if(!(i in d))print a[i]}' ./ssg-rhel7-ds.xml > ./ssg-rhel7-ds-1.3.xml`
- Validate it: `java -Xmx1024m -jar scapval-1.3.2.jar -file ./ssg-rhel7-ds-1.3.xml -scapversion 1.3 > scap-1.3.log`

- Report of scapval with workarounds: https://fedorapeople.org/~wsato/scap-1.3/validation-report.html

#### Rationale:

- SCAP 1.3 should supersede SCAP 1.2 
- In the near future OpenSCAP should be update to handle generation of SCAP 1.3 DS.
  - More specifically the command `oscap ds sds-compose`
